SpringKonstant/src/org/usfirst/frc6357/robotcode/subsystems/TFMini.java

```diff
@@ -183,15 +158,6 @@ private int amountOfBytes()
         return sensor.getBytesReceived();
     }
     
-    /**
-     * Read raw bytes out of the buffer
-     * @return An array of the read bytes
-     */
-    private byte[] readBytes()
-    {
-        return sensor.read(PACKET_LENGTH);
-    }
-    
     /**
      * Runs a loop in a Thread that will read the 9 byte packets and set each
      * byte to the correct piece of data for use if the data is valid.
```

Comment:
Do we want this to be a thread within the class, or should the class implement Thread? I can see some problems arising if we call start() or runLiDAR() more than once, as each call would spawn a new thread on the same object.

Reply:
It should definitely be a single instance thread but I don't know how to achieve this in Java. Each object instantiated should use the distance, strength and quality value read by the single thread. That also means that the packet member must be global too. I seem to remember there's some modifier you can add to the member definition that makes this occur. Any idea what it is?